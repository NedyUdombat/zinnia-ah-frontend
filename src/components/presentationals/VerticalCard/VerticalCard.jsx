import PropTypes from 'prop-types';
import React from 'react';

import { Card, Icon, Image } from 'semantic-ui-react';
import PopularAuthorItem from '../PopularAuthorItem/PopularAuthorItem';

// components

// styles
import './VerticalCard.scss';
import Title from '../Title/Title';
import Button from '../Button/Button';
import { DEFAULT_ARTICLE_IMAGE_URL } from '../../../utils/config';
import { Link } from 'react-router-dom';

const VerticalCard = ({ index, article }) => {
  return (
    <div className={`card-${index}`}>
      <Card>
        <Link to={`/read/${article.slug}`}>
          <Image
            src={article.imageThumbnail || DEFAULT_ARTICLE_IMAGE_URL}
            wrapped
            ui={true}
          />
        </Link>
        <Card.Content>
          <Card.Header>
            <Link to={`/read/${article.slug}`}>
              <Title content={article.title} className="title-avg" />
            </Link>
            <Button
              className="btn-transparent"
              type="button"
              value={
                <Icon
                  name="bookmark outline"
                  style={{
                    fontSize: '1.8rem',
                    margin: '0',
                  }}
                />
              }
            />
          </Card.Header>
          <Card.Description>
            {' '}
            <Link to={`/read/${article.slug}`} style={{ color: 'inherit' }}>
              {' '}
              {article.description}
            </Link>
          </Card.Description>
        </Card.Content>
        <div className="card-footer">
          <PopularAuthorItem
            key={'uiojklnmkopiuygh'}
            name={''}
            url={article.author.username}
            image={article.author.image}
            username={article.author.username}
            link={`/read/${article.slug}`}
            value={'Read More'}
          />
        </div>
      </Card>
    </div>
  );
};

export default VerticalCard;

VerticalCard.propTypes = {
  article: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired,
};
