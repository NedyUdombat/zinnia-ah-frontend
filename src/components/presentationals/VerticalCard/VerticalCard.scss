@import '../../../assets/variables';

.ui.card {
  border-radius: 0;
  box-shadow: none;
  width: 100%;
  margin-right: auto;
}
.ui.image {
  width: 100%;
}
.ui.image img {
  width: 100%;
  min-height :15rem;
  max-height: 15rem;
}

.ui.card:hover {
  cursor: pointer;
}

.ui.card > .content > .header {
  display: flex;
  font-family: $lato;
  justify-content: space-between;
  margin: 0;
}

.ui.card > :first-child {
  border-radius: 0;
}

.ui.card > .image > img {
  border-radius: 0;
  display: block;
  max-height: 12.5rem;
  min-height: 12.5rem;
  width: 100%;
}
