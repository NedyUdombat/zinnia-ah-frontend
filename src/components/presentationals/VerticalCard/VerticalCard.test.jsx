import React from 'react';
import { shallow } from 'enzyme';

import VerticalCard from './VerticalCard';

const props = {
  article: {
    imageThumbnail: 'fgbjm',
    title: 'wefds',
    description: 'werfderfde',
    author: {
      username: 'sdfgvds',
      image: 'wefdwed',
      id: 'wsvdcwedcwed',
    },
  },
  index: 1,
};
describe('Vertical Card', () => {
  it('should render correctly', () => {
    const component = shallow(<VerticalCard {...props} />);
    expect(component).toMatchSnapshot();
  });
});
