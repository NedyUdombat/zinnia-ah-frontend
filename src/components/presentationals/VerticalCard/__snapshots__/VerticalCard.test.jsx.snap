// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertical Card should render correctly 1`] = `
<div
  className="card-1"
>
  <Card>
    <Link
      to="/read/undefined"
    >
      <Image
        as="img"
        src="fgbjm"
        ui={true}
        wrapped={true}
      />
    </Link>
    <CardContent>
      <CardHeader>
        <Link
          to="/read/undefined"
        >
          <Title
            className="title-avg"
            content="wefds"
          />
        </Link>
        <Button
          className="btn-transparent"
          type="button"
          value={
            <Icon
              as="i"
              name="bookmark outline"
              style={
                Object {
                  "fontSize": "1.8rem",
                  "margin": "0",
                }
              }
            />
          }
        />
      </CardHeader>
      <CardDescription>
         
        <Link
          style={
            Object {
              "color": "#777",
            }
          }
          to="/read/undefined"
        >
           
          werfderfde
        </Link>
      </CardDescription>
    </CardContent>
    <div
      className="card-footer"
    >
      <PopularAuthorItem
        image="wefdwed"
        key="uiojklnmkopiuygh"
        link="/read/undefined"
        name=""
        url="sdfgvds"
        username="sdfgvds"
        value="Read More"
      />
    </div>
  </Card>
</div>
`;
